City-of-Helsinki/helsinki-paatokset
# PP-319 & PP-186: Eu cookie compliance and React & Share fixes

This PR updates most Hel.fi base modules to the newest version and enables the React & Share block. It also replaces the eu_cookie_compliance block's JS library to a custom one because the allowed cookie selections wouldn't work before because of block ID mismatches.

**To test**
- Checkout branch
- Update `.env.local`with the following new entries. Correct API keys can be found in the ticket: https://helsinkisolutionoffice.atlassian.net/browse/PP-186
```
REACT_AND_SHARE_APIKEY_EN=APIKEY
REACT_AND_SHARE_APIKEY_FI=APIKEY
REACT_AND_SHARE_APIKEY_SV=APIKEY
```
- Run `make up`
- Make sure the EU Cookie Compliance module gets patched by removing it first:
  - `rm -rf vendor/drupal/eu_cookie_compliance`
  - `rm -rf public/modules/contrib/eu_cookie_compliance/`
- Run `make new`
- Create a new page with the URL alias `/evasteasetukset`
- Open the site in a new private browsing window: https://helsinki-paatokset.docker.so/fi/evasteasetukset
- When you scroll down to the bottom of the page you should see the "React & Share" block disabled with a message saying it can't be shown until you accept cookies
- Accept all cookies.
  - The "React & Share" banner should be visible now
  - The cookie categories on the page should  be selected now. Try deselecting some of them and saving your settings
  - The settings should reflect the selections you made
- Try the same steps while logged in as an admin
